File: src/main/java/org/example/rxjava3/Rxjava3Application.java
package org.example.rxjava3;

import io.reactivex.rxjava3.core.Observable;
import io.reactivex.rxjava3.subjects.PublishSubject;

public class Rxjava3Application {
    public static void main(String[] args) {
        String abc = null;
        System.out.println("System".equals(abc));

        Observable
                .range(1, 20)
                .subscribe(System.out::println);

        var subject = PublishSubject.create();
        subject.onNext(1);
    }
}

File: src/main/kotlin/org/example/rxjava3/RxJava3Application.kt
package org.example.rxjava3

import io.reactivex.rxjava3.core.Observable
import io.reactivex.rxjava3.internal.schedulers.ExecutorScheduler
import io.reactivex.rxjava3.schedulers.Schedulers
import java.util.concurrent.Executors

fun main() {

    Observable.just(40)
        .subscribe {
            println("Thread ${Thread.currentThread().id} $it")
        }

    Observable.just(55)
        .subscribeOn(Schedulers.io())
        .observeOn(Schedulers.computation())

        .flatMap {
            println("Thread ${Thread.currentThread().name} flatMap")
            Observable.just(it.toString())
        }
//        .observeOn(Schedulers.computation())
//        .subscribeOn(Schedulers.from(Executors.newSingleThreadExecutor()))
        .subscribe {
            println("Thread ${Thread.currentThread().name} $it")
        }
}